File: max_value.js
function maxValue(array) {
    if (array.length == 0) {
        return null
    }
    if (array.length == 1) {
        return array[0]
    }
    else {
        var sub_max = maxValue(array.slice(1))
        console.log(sub_max);
        return array[0] > sub_max ? array[0] : sub_max
    }
}

console.log(maxValue([2,1,5,3,9,6]));

File: python/max_value.py
def maxValue(array):
    if len(array) == 0:
        return None
    if len(array) == 1:
        return array[0]
    else:
        sub_max = maxValue(array[1:])
        return array[0] if array[0] > sub_max else sub_max
print(maxValue([2,1,5,3,9,6]))